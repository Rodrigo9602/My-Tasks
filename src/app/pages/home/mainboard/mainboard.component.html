<div class="home-wrapper">
    <div class="sidebar">
        <header>
            <p class="header">Mainboard</p>
        </header>

        <div class="filters">

        </div>
    </div>
    <div class="content">
        <header>
            <nav>
                <p class="header">My Tasks</p>
                <div class="search-wrapper">
                    <div class="input-wrapper">
                        <input type="text" class="input-sec" placeholder="Search" name="taskName"
                            pattern="^[a-zA-Z0-90-9áéíóúÁÉÍÓÚñÑ\s]+$" #taskName="ngModel" [(ngModel)]="searchStr"
                            required="">

                        <button type="button" class="btn-search-icon" (click)="onSearch()"
                            [disabled]="taskName.invalid">
                            <fa-icon [icon]="searchIcon" />
                        </button>
                    </div>
                </div>
                <div class="options-wrapper">
                    <button class="btn-icon" (click)="onChangeMode()">
                        <fa-icon [icon]="modeIcon" />
                    </button>
                    <button class="btn-icon" (click)="onUserMenu()">
                        <fa-icon [icon]="userIcon" />
                    </button>

                    <div [ngClass]="{'user-menu-show': !userMenuHide , 'user-menu-hide': userMenuHide}">
                        <div class="menu-item">
                            <button class="btn-options">
                                <fa-icon [icon]="editIcon"></fa-icon>
                                <p>edit profile</p>
                            </button>
                        </div>

                        <div class="menu-item">
                            <button class="btn-options" (click)="onLogout()">
                                <fa-icon [icon]="logoutIcon"></fa-icon>
                                <p>log out</p>
                            </button>
                        </div>


                    </div>
                </div>

            </nav>
        </header>
        <div class="main-content">
            <ng-container *ngIf="user.name !== '' ">
                <app-task-table [userName]="user.name" (taskAdded)="onAddTask($event)" (taskRemoved)="onRemoveTask($event)"/>
            </ng-container>
        </div>
    </div>
    <div class="footer">
        <div class="row">
            <p>Created with </p><fa-icon [icon]="loveIcon" class="heart-icon"/><p> and </p><fa-icon [icon]="coffeIcon" class="coffee-icon"/>
        </div>
        
        
    </div>
</div>