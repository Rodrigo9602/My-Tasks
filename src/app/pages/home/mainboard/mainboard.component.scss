.home-wrapper {
    width: 100vw;
    min-height: 100vh;
    margin: 0;
    background-color: var(--bg);
    display: grid;
    grid-template-areas: 'sidebar main'
        'footer footer';
    grid-template-rows: 90% auto;
    grid-template-columns: 20% auto;
    gap: 0.5%;
    
}

.sidebar,
.content,
.footer {
    background-color: var(--items-bg);
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    border-radius: 1rem;
    padding: 10px 10px;
}

.sidebar {
    grid-area: sidebar;
    gap: 20px;
    

    header,
    .filters {
        border-radius: 0.5rem;
    }

    header {

        width: 90%;
        height: 60px;
        background-color: var(--bg);
        display: flex;
        align-items: center;
        justify-content: center;
    }

    p {
        color: var(--text-color);
    }

    .filters {
        width: 90%;
        background-color: var(--bg);
        display: flex;
        flex-direction: column;
        align-items: center;
        justify-content: space-around;

        p {
            font-family: 'poppins';
            font-size: 1rem;
        }

        .filters-wrapper {
            width: 90%;
            min-height: 440px;
            display: flex;
            flex-direction: column;
            align-items: center;
            justify-content: center;
            gap: 50px;
            

            mat-select {
                width: 90%;
                border-bottom: 1px solid var(--text-color);
                color: var(--text-color);
            }

            .btn-secondary {
                width: 90%;
                display: flex;
                align-items: center;
                justify-content: center;
                gap: 20px;
            }
        }
    }
}

.content {
    grid-area: main;
    gap: 20px;
    

    header,
    .main-content {
        border-radius: 0.5rem;
    }


    header {        
        width: 95%;
        height: 60px;
        background-color: var(--bg);
        

        nav {
            display: flex;
            height: 60px;
            align-items: center;
            justify-content: space-around;     
            padding: 0 20px;                    

            p {
                color: var(--text-color);
            }

            .btn-menu {
                display: hidden;
                width: 25px;
                height: 25px;
                transition: .5s ease-in-out;
                cursor: pointer;
                position: absolute;
                left: 8%;
            }

            .btn-menu,
            .btn-menu:focus,
            .btn-menu:active {
                outline: none;
                box-shadow: none;
                border: 0;
                background-color: transparent;
            }

            .btn-menu span {
                margin: 0;
                padding: 0;
            }

            .toggler-icon {
                height: 2px;
                width: 20px;
                border-radius: 5px;
                opacity: 1;
                transform: rotate(0deg);
                background-color: var(--text-color);
                transition: all .3s ease-in-out;
            }

            .top-bar {
                margin-top: 0px;
            }

            .middle-bar {
                margin-left: -10px;
                width: 10px;
            }

            .middle-bar,
            .bottom-bar {
                margin-top: 5px;
            }


            .top-bar-pressed {
                transform: rotate(315deg);
                position: absolute;
                top: 10px !important;

            }

            .middle-bar-pressed {
                opacity: 0;
                filter: alpha(opacity=0);
            }

            .bottom-bar-pressed {
                transform: rotate(45deg);
                margin-top: -8px !important;

            }

            .search-wrapper {                
                margin: 0 auto;
            }


            .input-sec {
                width: 35vw;
                height: 2.5rem;
                border-radius: 2rem;
                border: 1px solid var(--bg);
                font-size: 0.85rem;
                padding-left: 1.5rem;
                background-color: var(--items-bg);
                color: var(--text-color);
                font-family: 'poppins';
                outline: none;
                transition: all 0.3s ease;
            }



            .btn-search-icon {
                position: absolute;
                margin-left: 90%;
                background-color: var(--bg);
                color: var(--text-color);
                width: 2rem;
                height: 2rem;
                border: 1px solid transparent;
                border-radius: 50%;
                outline: none;
                font-size: 1.125rem;
                cursor: pointer;
                transition: all 0.3s ease;
            }

            .btn-search-icon:disabled {
                cursor: not-allowed;
            }

            .options-wrapper {
                display: flex;
                align-items: center;
                justify-content: center;
                gap: 1rem;
                position: absolute;
                right: 50px;

                .btn-icon {
                    font-size: 1.2rem;
                    width: 2.5rem;
                    height: 2.5rem;
                    border: 1px solid transparent;
                    border-radius: 50%;
                    cursor: pointer;
                    transition: all 0.3s ease;
                }

                .btn-icon:hover {
                    background-color: var(--items-bg);
                }

                .user-menu-show,
                .user-menu-hide {
                    position: absolute;
                    margin-top: 150px;
                    margin-left: 50px;
                    width: 10%;
                    min-width: 100px;
                    background-color: var(--bg);
                    color: var(--text-color);
                    box-shadow: rgba(0, 0, 0, 0.35) 0px 2px 5px;
                    border-radius: 0.5rem;
                    display: flex;
                    flex-direction: column;
                    align-items: center;
                    justify-content: center;
                    padding: 10px 20px;
                    z-index: -1;
                    transition: all 0.3s ease;

                    .menu-item {
                        width: 100%;
                        display: flex;
                        align-items: center;
                        justify-content: space-around;
                        font-family: 'poppins';
                    }
                }

                .user-menu-show::before,
                .user-menu-hide::before {
                    position: absolute;
                    content: "";
                    height: 15px;
                    width: 15px;
                    background: var(--bg);
                    top: -8px;
                    left: 52%;
                    transform: translate(-50%) rotate(45deg);
                    transition: all 0.3s cubic-bezier(0.68, -0.55, 0.265, 1.55);
                }

                .user-menu-show {
                    opacity: 1;
                    z-index: 5;
                }

                .user-menu-hide {
                    opacity: 0;
                }

            }



        }
        

    }

    mat-sidenav-container {
        width: 100%;
    }

    mat-sidenav-content {
        width: 100%;
        display: flex;
        align-items: center;
        justify-content: center;
        background-color: var(--items-bg);
    }


    .main-content {
        width:95%;    
        min-height: 480px;    
        background-color: var(--bg);
        display: flex;
        flex-direction: column;
        align-items: center;
        justify-content: center;
        gap: 20px;
        padding: 10px 0;


        header {
            width: 90%;
            height: 60px;
            display: flex;
            align-items: center;
            justify-content: space-between;            
            margin-top: 30px;

            p {
                font-family: 'poppins';
                color: var(--text-color);
            }

            .custom-primary {
                width: 12rem;
            }

            .btn-icon-custom {
                display: none;
            }


        }

        .table-wrapper {
            width: 100%;
            display: flex;
            align-items: center;
            justify-content: center;
        }
    }
}



.footer {
    grid-area: footer;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;

    .row {
        display: flex;
        align-items: center;
        justify-content: center;
        gap: 10px;

        p {
            font-family: 'poppins';
            color: var(--text-color);
        }

        .heart-icon {
            color: #dc2727;
        }

        .coffee-icon {
            color: #7c3030;
        }
    }


}





@media (max-width: 1000px) {

    .home-wrapper {
        grid-template-areas: 'main main'
            'footer footer';
    }

    .sidebar {
        display: none;
    }

    .content {
        .title {
            display: none;
        }

        .btn-menu {
            display: flex;
            flex-direction: column;
            align-items: center;
            justify-content: center;
            gap: 3px;
            margin-top: 5px;
            z-index: 2;
        }

        header {            
            width: 95%;   
            margin-top: 100px;         
            background-color: var(--bg);
        }

    }

    mat-sidenav {
        width: 30%;
        min-width: 300px;        


        header {
            width: 90%;
            height: 60px;
            background-color: var(--bg);
            display: flex;
            align-items: center;
            justify-content: center;
                       
        }
    
        p {
            color: var(--text-color);
        }
    
        .filters {
            width: 90%;
            background-color: var(--bg);
            display: flex;
            flex-direction: column;
            align-items: center;
            justify-content: center;
            gap:30px;
    
            p {
                font-family: 'poppins';
                font-size: 1rem;
            }
    
            .filters-wrapper {
                width: 90%;
                min-height: 440px;
                display: flex;
                flex-direction: column;
                align-items: center;
                justify-content: flex-start;
                gap: 50px;
    
                mat-select {
                    width: 90%;
                    border-bottom: 1px solid var(--text-color);
                    color: var(--text-color);
                }
    
                .btn-secondary {
                    width: 90%;
                    display: flex;
                    align-items: center;
                    justify-content: center;
                    gap: 20px;
                }
            }
        }
        
    }
   
}


@media (max-width: 675px) { 
    .content {
        header {
            nav {                
                .search-wrapper {
                    display: none;
                }
            }
        }
    }

    mat-sidenav {
        .search-wrapper {                
            margin: -200px auto 0 auto;
        }


        .input-sec {
            width: 250px;
            height: 2.5rem;
            border-radius: 2rem;
            border: 1px solid var(--bg);
            font-size: 0.85rem;
            padding-left: 1.5rem;
            background-color: var(--items-bg);
            color: var(--text-color);
            font-family: 'poppins';
            outline: none;
            transition: all 0.3s ease;
        }



        .btn-search-icon {
            position: absolute;
            margin-left: 70%;
            background-color: var(--bg);
            color: var(--text-color);
            width: 2rem;
            height: 2rem;
            border: 1px solid transparent;
            border-radius: 50%;
            outline: none;
            font-size: 1.125rem;
            cursor: pointer;
            transition: all 0.3s ease;
        }

        .btn-search-icon:disabled {
            cursor: not-allowed;
        }
    }
}


@media (max-width: 450px) {
    .content {
        .main-content {
            header {
                align-items: center;
                justify-content: center;
                gap: 50px;

                .custom-primary {
                    display: none;
                }

                .btn-icon-custom {
                    width: 25px;
                    height: 25px;
                    display: flex;
                    align-items: center;
                    justify-content: center;
                    color: var(--bg);
                    background-color: var(--items-color-gradient);                
                    border-radius: 50% !important;
                }
            }
        }
    }
}