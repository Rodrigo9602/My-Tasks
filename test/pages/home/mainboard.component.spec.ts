import { ComponentFixture, TestBed } from '@angular/core/testing';

import { MainboardComponent } from '../../../src/app/pages/home/mainboard/mainboard.component';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { HttpClientTestingModule } from '@angular/common/http/testing';


describe('MainboardComponent', () => {
  let component: MainboardComponent;
  let fixture: ComponentFixture<MainboardComponent>;
  let compiled: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [MainboardComponent, BrowserAnimationsModule, HttpClientTestingModule],      
    })
    .compileComponents();
    
    fixture = TestBed.createComponent(MainboardComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match initial snapshot', () => {
    expect(compiled).toMatchSnapshot();
  });
});
