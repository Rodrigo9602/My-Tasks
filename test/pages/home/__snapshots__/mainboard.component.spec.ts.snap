// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainboardComponent should match initial snapshot 1`] = `
<div
  id="root1"
>
  <div
    class="home-wrapper"
  >
    <div
      class="sidebar"
    >
      <header>
        <p
          class="header"
        >
          Mainboard
        </p>
      </header>
      <div
        class="filters"
      >
        <p>
          Tasks filtering panel
        </p>
        <div
          class="filters-wrapper"
        >
          <mat-select
            aria-autocomplete="none"
            aria-disabled="false"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-invalid="false"
            aria-required="false"
            class="mat-mdc-select mat-mdc-select-empty"
            name="taskState"
            ngskiphydration=""
            role="combobox"
            tabindex="0"
          >
            <div
              cdk-overlay-origin=""
              class="mat-mdc-select-trigger"
            >
              <div
                class="mat-mdc-select-value"
              >
                <span
                  class="mat-mdc-select-placeholder mat-mdc-select-min-line"
                >
                  
                </span>
                
                
              </div>
              <div
                class="mat-mdc-select-arrow-wrapper"
              >
                <div
                  class="mat-mdc-select-arrow"
                >
                  <svg
                    aria-hidden="true"
                    focusable="false"
                    height="24px"
                    viewBox="0 0 24 24"
                    width="24px"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            
          </mat-select>
          <button
            class="btn-secondary"
          >
            Show oldest to-do
          </button>
          <button
            class="btn-secondary"
          >
            <fa-icon
              class="ng-fa-icon"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-clock"
                data-icon="clock"
                data-prefix="far"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M464 256A208 208 0 1 1 48 256a208 208 0 1 1 416 0zM0 256a256 256 0 1 0 512 0A256 256 0 1 0 0 256zM232 120V256c0 8 4 15.5 10.7 20l96 64c11 7.4 25.9 4.4 33.3-6.7s4.4-25.9-6.7-33.3L280 243.2V120c0-13.3-10.7-24-24-24s-24 10.7-24 24z"
                  fill="currentColor"
                />
              </svg>
            </fa-icon>
             late tasks 
          </button>
        </div>
      </div>
    </div>
    <div
      class="content"
    >
      <header>
        <nav>
          <p
            class="header title"
          >
            My Tasks
          </p>
          <button
            class="btn-menu"
          >
            <span
              class="toggler-icon top-bar"
            />
            <span
              class="toggler-icon middle-bar"
            />
            <span
              class="toggler-icon bottom-bar"
            />
          </button>
          <div
            class="search-wrapper"
          >
            <div
              class="input-wrapper"
            >
              <input
                class="input-sec ng-untouched ng-pristine ng-invalid"
                name="taskName"
                pattern="^[a-zA-Z0-90-9áéíóúÁÉÍÓÚñÑ\\s]+$"
                placeholder="type a word..."
                required=""
                type="text"
              />
              <button
                class="btn-search-icon"
                disabled=""
                type="button"
              >
                <fa-icon
                  class="ng-fa-icon"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-magnifying-glass"
                    data-icon="magnifying-glass"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
                      fill="currentColor"
                    />
                  </svg>
                </fa-icon>
              </button>
            </div>
          </div>
          <div
            class="options-wrapper"
          >
            <button
              class="btn-icon"
            >
              <fa-icon
                class="ng-fa-icon"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-moon"
                  data-icon="moon"
                  data-prefix="far"
                  focusable="false"
                  role="img"
                  viewBox="0 0 384 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M144.7 98.7c-21 34.1-33.1 74.3-33.1 117.3c0 98 62.8 181.4 150.4 211.7c-12.4 2.8-25.3 4.3-38.6 4.3C126.6 432 48 353.3 48 256c0-68.9 39.4-128.4 96.8-157.3zm62.1-66C91.1 41.2 0 137.9 0 256C0 379.7 100 480 223.5 480c47.8 0 92-15 128.4-40.6c1.9-1.3 3.7-2.7 5.5-4c4.8-3.6 9.4-7.4 13.9-11.4c2.7-2.4 5.3-4.8 7.9-7.3c5-4.9 6.3-12.5 3.1-18.7s-10.1-9.7-17-8.5c-3.7 .6-7.4 1.2-11.1 1.6c-5 .5-10.1 .9-15.3 1c-1.2 0-2.5 0-3.7 0c-.1 0-.2 0-.3 0c-96.8-.2-175.2-78.9-175.2-176c0-54.8 24.9-103.7 64.1-136c1-.9 2.1-1.7 3.2-2.6c4-3.2 8.2-6.2 12.5-9c3.1-2 6.3-4 9.6-5.8c6.1-3.5 9.2-10.5 7.7-17.3s-7.3-11.9-14.3-12.5c-3.6-.3-7.1-.5-10.7-.6c-2.7-.1-5.5-.1-8.2-.1c-3.3 0-6.5 .1-9.8 .2c-2.3 .1-4.6 .2-6.9 .4z"
                    fill="currentColor"
                  />
                </svg>
              </fa-icon>
            </button>
            <button
              class="btn-icon"
            >
              <fa-icon
                class="ng-fa-icon"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-circle-user"
                  data-icon="circle-user"
                  data-prefix="far"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M406.5 399.6C387.4 352.9 341.5 320 288 320H224c-53.5 0-99.4 32.9-118.5 79.6C69.9 362.2 48 311.7 48 256C48 141.1 141.1 48 256 48s208 93.1 208 208c0 55.7-21.9 106.2-57.5 143.6zm-40.1 32.7C334.4 452.4 296.6 464 256 464s-78.4-11.6-110.5-31.7c7.3-36.7 39.7-64.3 78.5-64.3h64c38.8 0 71.2 27.6 78.5 64.3zM256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zm0-272a40 40 0 1 1 0-80 40 40 0 1 1 0 80zm-88-40a88 88 0 1 0 176 0 88 88 0 1 0 -176 0z"
                    fill="currentColor"
                  />
                </svg>
              </fa-icon>
            </button>
            <div
              class="user-menu-hide"
            >
              <div
                class="menu-item"
              >
                <button
                  class="btn-options"
                >
                  <fa-icon
                    class="ng-fa-icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-right-from-bracket"
                      data-icon="right-from-bracket"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 512 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M377.9 105.9L500.7 228.7c7.2 7.2 11.3 17.1 11.3 27.3s-4.1 20.1-11.3 27.3L377.9 406.1c-6.4 6.4-15 9.9-24 9.9c-18.7 0-33.9-15.2-33.9-33.9l0-62.1-128 0c-17.7 0-32-14.3-32-32l0-64c0-17.7 14.3-32 32-32l128 0 0-62.1c0-18.7 15.2-33.9 33.9-33.9c9 0 17.6 3.6 24 9.9zM160 96L96 96c-17.7 0-32 14.3-32 32l0 256c0 17.7 14.3 32 32 32l64 0c17.7 0 32 14.3 32 32s-14.3 32-32 32l-64 0c-53 0-96-43-96-96L0 128C0 75 43 32 96 32l64 0c17.7 0 32 14.3 32 32s-14.3 32-32 32z"
                        fill="currentColor"
                      />
                    </svg>
                  </fa-icon>
                  <p>
                    log out
                  </p>
                </button>
              </div>
            </div>
          </div>
        </nav>
      </header>
      <mat-sidenav-container
        class="mat-drawer-container mat-sidenav-container"
        ngskiphydration=""
      >
        <div
          class="mat-drawer-backdrop"
        />
        
        <div
          aria-hidden="true"
          class="cdk-visually-hidden cdk-focus-trap-anchor"
          tabindex="0"
        />
        <mat-sidenav
          class="mat-drawer mat-sidenav ng-trigger ng-trigger-transform mat-drawer-over"
          ngskiphydration=""
          style="box-shadow: none; visibility: hidden;"
          tabindex="-1"
        >
          <div
            cdkscrollable=""
            class="mat-drawer-inner-container"
          >
            <header>
              <p
                class="header"
              >
                Mainboard
              </p>
            </header>
            <div
              class="filters"
            >
              <p>
                Tasks filtering panel
              </p>
              <div
                class="filters-wrapper"
              >
                <mat-select
                  aria-autocomplete="none"
                  aria-disabled="false"
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-invalid="false"
                  aria-required="false"
                  class="mat-mdc-select mat-mdc-select-empty"
                  name="taskState"
                  ngskiphydration=""
                  role="combobox"
                  style=""
                  tabindex="0"
                >
                  <div
                    cdk-overlay-origin=""
                    class="mat-mdc-select-trigger"
                  >
                    <div
                      class="mat-mdc-select-value"
                    >
                      <span
                        class="mat-mdc-select-placeholder mat-mdc-select-min-line"
                      >
                        
                      </span>
                      
                      
                    </div>
                    <div
                      class="mat-mdc-select-arrow-wrapper"
                    >
                      <div
                        class="mat-mdc-select-arrow"
                      >
                        <svg
                          aria-hidden="true"
                          focusable="false"
                          height="24px"
                          viewBox="0 0 24 24"
                          width="24px"
                        >
                          <path
                            d="M7 10l5 5 5-5z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                  
                </mat-select>
                <button
                  class="btn-secondary"
                >
                  Show oldest to-do
                </button>
                <button
                  class="btn-secondary"
                >
                  <fa-icon
                    class="ng-fa-icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-clock"
                      data-icon="clock"
                      data-prefix="far"
                      focusable="false"
                      role="img"
                      viewBox="0 0 512 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M464 256A208 208 0 1 1 48 256a208 208 0 1 1 416 0zM0 256a256 256 0 1 0 512 0A256 256 0 1 0 0 256zM232 120V256c0 8 4 15.5 10.7 20l96 64c11 7.4 25.9 4.4 33.3-6.7s4.4-25.9-6.7-33.3L280 243.2V120c0-13.3-10.7-24-24-24s-24 10.7-24 24z"
                        fill="currentColor"
                      />
                    </svg>
                  </fa-icon>
                   late tasks 
                </button>
              </div>
            </div>
            <div
              class="search-wrapper"
            >
              <div
                class="input-wrapper"
              >
                <input
                  class="input-sec ng-untouched ng-pristine ng-invalid"
                  name="taskName"
                  pattern="^[a-zA-Z0-90-9áéíóúÁÉÍÓÚñÑ\\s]+$"
                  placeholder="type a word..."
                  required=""
                  type="text"
                />
                <button
                  class="btn-search-icon"
                  disabled=""
                  type="button"
                >
                  <fa-icon
                    class="ng-fa-icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-magnifying-glass"
                      data-icon="magnifying-glass"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 512 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
                        fill="currentColor"
                      />
                    </svg>
                  </fa-icon>
                </button>
              </div>
            </div>
          </div>
        </mat-sidenav>
        <div
          aria-hidden="true"
          class="cdk-visually-hidden cdk-focus-trap-anchor"
          tabindex="0"
        />
        <mat-sidenav-content
          class="mat-drawer-content mat-sidenav-content"
          ngskiphydration=""
        >
          <div
            class="main-content"
          >
            
          </div>
          
          
        </mat-sidenav-content>
        
      </mat-sidenav-container>
    </div>
    <div
      class="footer"
    >
      <div
        class="row"
      >
        <p>
          Created with 
        </p>
        <fa-icon
          class="ng-fa-icon heart-icon"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-heart"
            data-icon="heart"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M47.6 300.4L228.3 469.1c7.5 7 17.4 10.9 27.7 10.9s20.2-3.9 27.7-10.9L464.4 300.4c30.4-28.3 47.6-68 47.6-109.5v-5.8c0-69.9-50.5-129.5-119.4-141C347 36.5 300.6 51.4 268 84L256 96 244 84c-32.6-32.6-79-47.5-124.6-39.9C50.5 55.6 0 115.2 0 185.1v5.8c0 41.5 17.2 81.2 47.6 109.5z"
              fill="currentColor"
            />
          </svg>
        </fa-icon>
        <p>
           and 
        </p>
        <fa-icon
          class="ng-fa-icon coffee-icon"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-mug-saucer"
            data-icon="mug-saucer"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 640 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M96 64c0-17.7 14.3-32 32-32H448h64c70.7 0 128 57.3 128 128s-57.3 128-128 128H480c0 53-43 96-96 96H192c-53 0-96-43-96-96V64zM480 224h32c35.3 0 64-28.7 64-64s-28.7-64-64-64H480V224zM32 416H544c17.7 0 32 14.3 32 32s-14.3 32-32 32H32c-17.7 0-32-14.3-32-32s14.3-32 32-32z"
              fill="currentColor"
            />
          </svg>
        </fa-icon>
      </div>
    </div>
  </div>
</div>
`;
